Modules/_pickle.c

```diff
@@ -952,6 +939,33 @@ _Pickler_FlushToFile(PicklerObject *self)
     return (result == NULL) ? -1 : 0;
 }
 
+static int
+_Pickler_OpcodeBoundary(PicklerObject *self)
+{
+    Py_ssize_t frame_len;
+
+    if (!self->framing || self->frame_start == -1)
+        return 0;
+    frame_len = self->output_len - self->frame_start - FRAME_HEADER_SIZE;
+    if (frame_len >= FRAME_SIZE_TARGET) {
+        if(_Pickler_CommitFrame(self, 0)) {
+            return -1;
+        }
+        if (self->write != NULL) {
```

Comment:
Should this block be part of `_Pickler_CommitFrame` instead?

Reply:
Actually `_Pickler_FlushToFile` calls `_Pickler_GetString` which calls `_Pickler_CommitFrame` (that does nothing if the frame has already been committed).

We always need to `_Pickler_CommitFrame` here, but flushing should only be done if we are writing to a file (which is not always the case, e.g. when calling `dumps`).

I could refactor those three functions but I was afraid to introduce subtle semantic behaviors.